includes/class-wc-payments-payment-request-button-handler.php

```diff
@@ -862,15 +862,15 @@ public function ajax_get_shipping_options() {
 		$shipping_address          = filter_input_array(
 			INPUT_POST,
 			[
-				'country'   => FILTER_SANITIZE_STRING,
-				'state'     => FILTER_SANITIZE_STRING,
-				'postcode'  => FILTER_SANITIZE_STRING,
-				'city'      => FILTER_SANITIZE_STRING,
-				'address_1' => FILTER_SANITIZE_STRING,
-				'address_2' => FILTER_SANITIZE_STRING,
+				'country'   => FILTER_SANITIZE_SPECIAL_CHARS,
+				'state'     => FILTER_SANITIZE_SPECIAL_CHARS,
+				'postcode'  => FILTER_SANITIZE_SPECIAL_CHARS,
+				'city'      => FILTER_SANITIZE_SPECIAL_CHARS,
+				'address_1' => FILTER_SANITIZE_SPECIAL_CHARS,
+				'address_2' => FILTER_SANITIZE_SPECIAL_CHARS,
```

Comment:
I was wondering that since we are using the same sanitisation function for all parameters here, why should we remain beholden to this `filter_input_array` function? We could just rewrite this logic ourselves: e.g. try to `array_filter` the `$_POST` object by accepted keys, then `array_map` with `sanitize_text_field` or something like that...but in the end AFAICT that wouldn't really offer that much in the way of any particular performance advantage and there isn't really anything particularly pernicious about this current approach with using `filter_input_array`, so it's probably not worth getting hot and bothered over.

Anyways, I don't really have any recommendations with this comment, just thinking out loud, so don't mind me. 😅

Reply:
I agree and also agree this wouldn't really change things a lot, so I keep things as they are 👍 Thanks for sharing your thoughts!